padlet/padlet-qrcode-old
# Update `svg` logic to render rounded dots, rounded corners, and an optional center image

Resolves [PRO-16969](https://linear.app/padlet/issue/PRO-16969/update-rqrcode-library-logic)

## 👀 Reviewers

Main reviewer: @julianalmandos 

## 🔎 What / why
This PR update QR code svg logic to follow this design https://www.figma.com/design/M03sUnd8Yngxu8eUINNhyn/Branded-QR-Codes?node-id=15-2971&m=dev.

Note: The implementation doesn't need to be pixel-perfect. As long as the QR code looks similar to the design, it should be acceptable.

## 🧪 Testing
Steps:
1. Open a terminal, open rails console (run `./bin/console`)
2. Run `puts RQRCode::QRCode.new("https://padlet.com").as_svg({ image_url: "https://padlet.net/logos/crane_black_with_14px_white_stroke.png" })`
3. Copy the printed value and save as svg
4. The svg result should:
   - have logo
   - the corners and dots should be rounded

## 🖼 Screen captures

<details>

<summary>Screen captures</summary>

| Before | After |
| ------ | ----- |
|![CleanShot 2024-11-25 at 15 29 31@2x](https://github.com/user-attachments/assets/69fde234-9f27-4f65-91ea-76cba9a5c870)|![CleanShot 2024-11-25 at 15 28 59@2x](https://github.com/user-attachments/assets/278e86f4-bcda-4755-80bf-a0555279b245)|

</details>
